Code/C++/Doubly_linkedlist/Doubly_linkedlist_all_operations.cpp

```diff
@@ -0,0 +1,390 @@
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of node ";
+    cin >> new_node->data;
+    if (last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else
+    {
+        new_node->next = nullptr;
+        last->next = new_node;
+        new_node->prev = last;
+        last = new_node;
+    }
+}
+
+// Function for inserting new node next to given position(index) in list.
+
+void Doubly_linklist::insertion_after_given_index()
+{
+    node *ptr = start;
+    node *new_node = new node;
+    cout << "\nEnter the info of the node you want to insert";
+    cin >> new_node->data;
+    int count = 0, index;
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (index == 4)
+    {
+        insertion_at_end();
+    }
+    else
+    {
+        while (count != index)
+        {
+            ptr = ptr->next;
+            count += 1;
+        }
+        new_node->next = ptr->next;
+        ptr->next->prev = new_node;
+        ptr->next = new_node;
+        new_node->prev = ptr;
+    }
+}
+
+//Function to insert a node at given index.
+
+void Doubly_linklist::insertion_at_index()
+{
+    node *new_node = new node;
+    node *ptr = start;
+    cout << "\nEnter the info of the node you want to insert";
+    cin >> new_node->data;
+    int count = 0, index;
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (index == 0)
+    {
+        insertion_at_beginning();
+    }
+    else
+    {
+        while (count != index - 1)
+        {
+            ptr = ptr->next;
+            count += 1;
+        }
+        new_node->next = ptr->next;
+        ptr->next->prev = new_node;
+        ptr->next = new_node;
+        new_node->prev = ptr;
+    }
+}
+
+//Fuction for deleting a node from beginning.
+
+void Doubly_linklist::deletion_at_beginning()
+{
+    node *ptr = start;
+    if (start == nullptr && last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (start->next == nullptr)
+    {
+        start = nullptr;
+        last = nullptr;
+        free(start);
+        free(last);
+    }
+    else
+    {
+        start = start->next;
+        start->prev = nullptr;
+        ptr->next = nullptr;
+    }
+}
+
+//Fuction for deleting a node from end.
+
+void Doubly_linklist::deletion_at_end()
+{
+    node *temp = last;
+    if (start == nullptr && last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (start->next == nullptr)
+    {
+        start = nullptr;
+        last = nullptr;
+        free(start);
+        free(last);
```

Comment:
free the memory before assigning then the value of nullptr. Else, there will be a memory leak.